src/gepa/api.py

```diff
@@ -29,6 +29,8 @@ def optimize(
     skip_perfect_score=True,
     reflection_minibatch_size=3,
     perfect_score=1,
+    # Component selection configuration
+    module_selector: "ReflectionComponentSelector | None" = None,
```

Comment:
Let's make this "ReflectionComponentSelector | str | None"

Such that there are a few default component selectors that GEPA uses by default.

For example, look at candidate_selection_strategy.

This allows it to be very easily specified in command line.

Reply:
A much more scalable way is to have some kind of a "registry", which is basically a dict() mapping these strings to the classes. I don't think that needs to be part of this PR. There will be a much larger refactoring of the codebase to centralize all kinds of configs which is currently split across all files.